src/python/test/test_dxclient.py

```diff
@@ -963,6 +965,64 @@ def _test_dx_ssh(self, project, instance_type):
+                                                        inputSpec=[], outputSpec=[],
+                                                        dxapi="1.0.0", version="1.0.0",
+                                                        project=self.project))["id"]
+                sleep_applet2 = dxpy.api.applet_new(dict(name="sleep2",
+                                                        runSpec={"code": "sleep 5",
+                                                                 "interpreter": "bash",
+                                                                 "distribution": "Ubuntu", "release": "14.04",
+                                                                 "execDepends": [{"name": "dx-toolkit"}],
+                                                                 "systemRequirements": {"*": {"instanceType": "mem2_hdd2_x1"}}},
+                                                        inputSpec=[], outputSpec=[],
+                                                        dxapi="1.0.0", version="1.0.0",
+                                                        project=self.project))["id"]
+
+                dx = pexpect.spawn("dx run {} --yes --ssh".format(sleep_applet1),
+                                   env=override_environment(HOME=wd),
+                                   **spawn_extra_args)
+                dx.logfile = sys.stdout
+                dx.setwinsize(20, 90)
+                dx.expect("Waiting for job")
+                dx.expect("Resolving job hostname and SSH host key", timeout=1200)
+
+                dx.expect("This is the DNAnexus Execution Environment", timeout=600)
+                # Check for job name (e.g. "Job: sleep")
+                #dx.expect("Job: \x1b\[1msleep", timeout=5)
+                if USING_PYTHON2:
+                    # \xf6 is ö
+                    project_line = "Project: dxclient_test_pr\xf6ject".encode(sys_encoding)
+                else:
+                    project_line = "Project: dxclient_test_pröject"
+                dx.expect(project_line)
+
+                dx.expect("The job is running in terminal 1.", timeout=5)
+                # Check for terminal prompt and verify we're in the container
+                job = next(dxpy.find_jobs(name="sleep", project=self.project), None)
+                job_id = job['id']
+                dx.expect("OS version: Ubuntu 14.04", timeout=5)
+                dx.sendline("dx run {} --yes --detach".format(sleep_applet2))
+                time.sleep(1)
+                job2 = next(dxpy.find_jobs(name="sleep2", project=self.project), None)
+                print(job2)
+                self.assertTrue(job_id in job2['detachedFrom'])
+                dx.sendline("exit")
+                dx.expect("bash running", timeout=10)
+                dx.sendcontrol("c")  # CTRL-c
+                dx.expect("[exited]")
+                dx.expect("dnanexus@job", timeout=10)
```

Comment:
Thanks for adding the test! It's going to be super useful (the pexpect experience) for our other interactive apps. 

Please:
* terminate both jobs at the end, with something like: run("dx terminate " + job_id)
* remove print

Reply:
As discussed, I'll leave this test out for now - it will take longer than expected

Reply:
SG, thanks!